keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,391 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Namespaces are not gauranteed to have any 'identifying' labels by the Kubernetes API.  This means that to select a namespace by name, on would have to use a field selector.  This encumberance makes it hard to write default NetworkPolicies, and also makes its tricky to create other kinds of "label-driven" namespace functionality in the K8s api.
+
+The idea of this KEP is to propose that namespaces are always ensured, by the API server, to have a *immutable label matching the metadata.name of the namespace*, which thus allows one to build label selectors to include/exclude namespaces by name, without having write access to a namespace (which is required to label it manually). 
+
+This was initially envisioned as a workaround to a limitation in the network policy API, but has emerged as a potential to other problems which may be solved by default namespace labels.
+
+## Motivation
+
+In https://github.com/kubernetes/enhancements/issues/2112 , we discussed several 
+ways to implement network policies that select namespaces by name, rather than by 
+relying on labels which may be unknown, or uneditable by people writing policies, 
+or untrustworthy.
+
+This was inspired by a broad community ask to target namespaces in this manner, 
+as evidenced by issues such as [kubernetes #88253](https://github.com/kubernetes/kubernetes/issues/88253), 
+and the larger set of user feedback obtained in  [user stories](https://github.com/jayunit100/network-policy-subproject/blob/master/p0_user_stories.md).
+
+It’s also worth noting that there are at least 2 other NamespaceSelector APIs in 
+k/k - Validating and Mutating webhooks.  They will inevitably need similar API 
+evolution, but we don't need to do that as part of this KEP.
+
+It turns out that alternate [solutions](#alternatives):
+Break the old API in fundamental ways that are questionably WRT cost/benefit OR
+add nesting complexity to the network policy API, wherein certain “subsets” of
+the API were not usable together.  This is not “bad”, but it is suboptimal and
+introduces redundant semantics. This is avoidable if we could assume, in all
+cases, that namespaces have a default label, thereby leveraging existing
+functionalities offered by label selectors.
+
+There may also be many other usecases where the security or permissions boundary
+might be conveniently defined in terms of a namepsace name, and that such a
+definition would be most easily referenced by a default label.
+
+### Goals
+
+- Add an immutable default label to ALL namespaces, as the namespace name so this 
+can be used as selector by arbitrary components.
```

Comment:
nit: This is more a summary of the implementation. I would say the goal is to enable things that select namespaces by labels can securely and reliably also select individual namespaces by name.